source-code/core/app/src/createInlang.ts

```diff
@@ -0,0 +1,205 @@
+import type { InlangInstance } from "./api.js"
+import { ImportFunction, ResolveModulesFunction, resolveModules } from "@inlang/module"
+import {
+	NodeishFilesystemSubset,
+	InlangConfig,
+	createQuery,
+	ResolvedPlugins,
+	Message,
+	tryCatch,
+	Plugin,
+} from "@inlang/plugin"
+import { TypeCompiler } from "@sinclair/typebox/compiler"
+import { Value } from "@sinclair/typebox/value"
+import { InvalidConfigError } from "./errors.js"
+import { LintError, LintReport, LintRule, lintMessages } from "@inlang/lint"
+import { createRoot, createSignal, createEffect } from "./solid.js"
+
+const ConfigCompiler = TypeCompiler.Compile(InlangConfig)
+
+/**
+ * Creates an inlang instance.
+ *
+ * - Use `_import` to pass a custom import function for testing,
+ *   and supporting legacy modules such as CJS.
+ *
+ */
+export const createInlang = async (args: {
+	configPath: string
```

Comment:
@felixhaeberle @ivanhofer @NilsJacobsen @NiklasBuchfink

I think createInlang should figure out itself where the config is located and should only take the workingDirectory as argument. 

- apps wouldn't need own "where is the config logic" 
- all apps behave the same way/find the same config upon opening a repository (and defining the working directory) 

Agree? Vote pls.

Reply:
Speaking about the CLI, how does it then work with self-defined --config paths in monorepo setups atm?

Reply:
We could make it optional and if set, the `resolveConfig` logic tries that file first

Reply:
> Speaking about the CLI, how does it then work with self-defined --config paths in monorepo setups atm? 

Set the working directory (which you can pass to createInlang) to the config path.